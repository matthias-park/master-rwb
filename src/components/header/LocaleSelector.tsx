import clsx from 'clsx';
import React from 'react';
import Dropdown from 'react-bootstrap/Dropdown';

interface Props {
  available: string[];
  current: string;
  setLocale: (locale: string) => void;
}

const LocaleSelector = ({ available, current, setLocale }: Props) => (
  <>
    <Dropdown className="header__nav-item">
      <Dropdown.Toggle
        data-testid="dropdown-trigger"
        variant="link"
        className="header__nav-item-link d-none d-xl-block"
        id="navbarDropdownMenuLink"
      >
        <strong data-testid="current" className="text-uppercase">
          {current}
        </strong>
      </Dropdown.Toggle>
      <Dropdown.Menu data-testid="available-desktop">
        {available.map(lang => (
          <Dropdown.Item
            data-testid={`desktop-locale-${lang}`}
            className={`lang-${lang} header__nav-item-link text-uppercase cursor-pointer`}
            key={lang}
            as="div"
            onClick={() => {
              setLocale(lang);
            }}
          >
            {lang}
          </Dropdown.Item>
        ))}
      </Dropdown.Menu>
    </Dropdown>
    <li className="d-flex d-xl-none languages" data-testid="available-mobile">
      {available.map(lang => (
        <div
          className={clsx(
            `lang-${lang}`,
            'header__nav-item-link text-uppercase cursor-pointer',
            lang === current && 'font-weight-bold',
          )}
          key={lang}
          onClick={() => {
            setLocale(lang);
          }}
        >
          {lang}
        </div>
      ))}
    </li>
  </>
);

export default LocaleSelector;
