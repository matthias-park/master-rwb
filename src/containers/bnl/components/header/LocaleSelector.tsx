import clsx from 'clsx';
import React, { useState } from 'react';
import Dropdown from 'react-bootstrap/Dropdown';
import { AvailableLocale } from '../../../../types/api/PageConfig';
import LoadingSpinner from '../../../../components/LoadingSpinner';

interface Props {
  available: AvailableLocale[];
  current: string;
  setLocale: (locale: string) => Promise<void>;
}

const LocaleSelector = ({ available, current, setLocale }: Props) => {
  const [updatingLocale, setUpdatingLocale] = useState('');
  const changeLocale = async (lang: string) => {
    if (!!updatingLocale) return;
    setUpdatingLocale(lang);
    await setLocale(lang);
    setUpdatingLocale('');
  };
  return (
    <>
      <Dropdown className="header__nav-item">
        <Dropdown.Toggle
          data-testid="dropdown-trigger"
          variant="link"
          className="header__nav-item-link d-none d-xl-block"
          id="navbarDropdownMenuLink"
          disabled={!!updatingLocale}
        >
          <LoadingSpinner show={!!updatingLocale} className="mr-1" />
          <strong data-testid="current" className="text-uppercase">
            {current}
          </strong>
        </Dropdown.Toggle>
        <Dropdown.Menu data-testid="available-desktop">
          {available.map(lang => (
            <Dropdown.Item
              data-testid={`desktop-locale-${lang.iso}`}
              className={`lang-${lang.iso} header__nav-item-link text-uppercase cursor-pointer`}
              key={lang.id}
              as="div"
              onClick={() => {
                changeLocale(lang.iso);
              }}
            >
              {lang.iso}
            </Dropdown.Item>
          ))}
        </Dropdown.Menu>
      </Dropdown>
      <li className="d-flex d-xl-none languages" data-testid="available-mobile">
        {available.map(lang => (
          <div
            className={clsx(
              `lang-${lang.iso}`,
              'header__nav-item-link text-uppercase cursor-pointer',
              ((lang.iso === current && !updatingLocale) ||
                lang.iso === updatingLocale) &&
                'font-weight-bold',
            )}
            key={lang.id}
            onClick={() => {
              changeLocale(lang.iso);
            }}
          >
            <LoadingSpinner
              show={updatingLocale === lang.iso}
              small
              className="mr-1"
            />
            {lang.iso}
          </div>
        ))}
      </li>
    </>
  );
};

export default LocaleSelector;
