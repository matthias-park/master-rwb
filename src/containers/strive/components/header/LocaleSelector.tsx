import clsx from 'clsx';
import React, { useState } from 'react';
import Dropdown from 'react-bootstrap/Dropdown';
import { AvailableLocale } from '../../../../types/api/PageConfig';
import LoadingSpinner from '../../../../components/LoadingSpinner';
import { StyledHeaderNavItem } from '../styled/StyledHeader';

interface Props {
  available: AvailableLocale[];
  current: string;
  setLocale: (locale: string) => Promise<void>;
}

const LocaleSelector = ({ available, current, setLocale }: Props) => {
  const [updatingLocale, setUpdatingLocale] = useState('');
  const changeLocale = async (lang: string) => {
    if (!!updatingLocale || lang === current) return;
    setUpdatingLocale(lang);
    await setLocale(lang);
    setUpdatingLocale('');
  };
  return (
    <>
      <Dropdown as={StyledHeaderNavItem}>
        <Dropdown.Toggle
          data-testid="dropdown-trigger"
          variant="link"
          className="nav-link d-none d-xl-block"
          id="navbarDropdownMenuLink"
          disabled={!!updatingLocale}
        >
          <LoadingSpinner small show={!!updatingLocale} className="mr-1" />
          <strong data-testid="current" className="text-uppercase">
            {current}
          </strong>
        </Dropdown.Toggle>
        <Dropdown.Menu data-testid="available-desktop">
          {available.map(lang => (
            <Dropdown.Item
              data-testid={`desktop-locale-${lang.iso}`}
              className={clsx(
                `lang-${lang.iso}`,
                'nav-link text-uppercase',
                lang.iso !== current && 'cursor-pointer',
              )}
              key={lang.id}
              as="div"
              onClick={() => {
                changeLocale(lang.iso);
              }}
            >
              {lang.iso}
            </Dropdown.Item>
          ))}
        </Dropdown.Menu>
      </Dropdown>
      <li className="d-flex d-xl-none languages" data-testid="available-mobile">
        {available.map(lang => (
          <div
            className={clsx(
              `lang-${lang.iso}`,
              'nav-link text-uppercase cursor-pointer',
              ((lang.iso === current && !updatingLocale) ||
                lang.iso === updatingLocale) &&
                'font-weight-bold',
            )}
            key={lang.id}
            onClick={() => {
              changeLocale(lang.iso);
            }}
          >
            <LoadingSpinner
              show={updatingLocale === lang.iso}
              small
              className="mr-1"
            />
            {lang.iso}
          </div>
        ))}
      </li>
    </>
  );
};

export default LocaleSelector;
