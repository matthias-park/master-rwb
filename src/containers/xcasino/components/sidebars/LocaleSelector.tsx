import React, { useState } from 'react';
import LeftSidebarMenu from './LeftSidebarMenu';
import { useConfig } from '../../../../hooks/useConfig';

interface LocaleSelectorProps {
  goBack: () => void;
}

const LocaleSelector = ({ goBack }: LocaleSelectorProps) => {
  const [updatingLocale, setUpdatingLocale] = useState('');
  const { locales, locale } = useConfig();

  const changeLocale = async (lang: string) => {
    /*if (!!updatingLocale || lang === locale) return;
    setUpdatingLocale(lang);
    await setLocale(lang);
    setUpdatingLocale('');*/
  };

  return (
    <LeftSidebarMenu
      goBack={goBack}
      items={locales.map(locale => ({
        title: locale.iso,
        onClick: () => changeLocale(locale.iso),
      }))}
    />
  );
};

export default LocaleSelector;
