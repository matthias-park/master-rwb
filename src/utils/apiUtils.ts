import fetch from 'isomorphic-unfetch';
import { ConfigInterface, mutate } from 'swr';
import { RailsApiResponseFallback } from '../constants';
import RailsApiResponse from '../types/api/RailsApiResponse';
import { RegistrationPostalCodeAutofill } from '../types/api/user/Registration';
import { cleanPostBody } from '.';
import * as Sentry from '@sentry/react';

export const formatUrl = (
  url: string,
  params: { [key: string]: string | number | undefined } = {},
): string => {
  const paramsArr = Object.keys(params)
    .filter(key => params[key])
    .map(key => `${key}=${params[key]}`);
  return `${url}${paramsArr.length ? '?' : ''}${paramsArr.join('&')}`;
};

interface GetApiOptions {
  responseText?: boolean;
  cache?: RequestCache;
}

export const getApi = <T>(url: string, options?: GetApiOptions): Promise<T> => {
  const config: RequestInit = {
    mode: 'cors',
    credentials: 'include',
    cache: options?.cache,
    headers: new Headers(),
  };
  const getUrl = url.startsWith('http')
    ? url
    : `${window.__config__.apiUrl}${url}`;
  return fetch(getUrl, config).then(res => {
    if (!res.ok && res.status !== 400) {
      if (![401, 403].includes(res.status)) {
        Sentry.captureMessage(
          `Request failed ${url} with status ${res.status}`,
          Sentry.Severity.Fatal,
        );
      } else {
        mutate('/restapi/v1/user/status');
      }
      return Promise.reject<RailsApiResponse<null>>({
        ...RailsApiResponseFallback,
        Code: res.status,
      });
    }
    if (options?.responseText) {
      return res.text();
    }
    return res.json();
  });
};

export interface PostOptions {
  formData?: boolean;
}

export const postApi = <T>(
  url: string,
  body?: {
    [key: string]: number | string | Blob | boolean | undefined | null | object;
  },
  options: PostOptions = {},
): Promise<T> => {
  const config: RequestInit = {
    method: 'post',
    mode: 'cors',
    credentials: 'include',
    headers: new Headers(),
  };
  if (body) {
    if (options.formData) {
      const formData = new FormData();
      const formBody = cleanPostBody(body);
      for (const key in formBody) {
        if (Array.isArray(formBody[key])) {
          for (const item of formBody[key]) {
            formData.append(key, item as string | Blob);
          }
        } else {
          formData.append(key, formBody[key] as string | Blob);
        }
      }
      config.body = formData;
    } else {
      config.body = JSON.stringify(cleanPostBody(body));
      (config.headers as Headers).append('Content-Type', 'application/json');
    }
  }
  const postUrl = url.startsWith('http')
    ? url
    : `${window.__config__.apiUrl}${url}`;
  return fetch(postUrl, config).then(res => {
    if (!res.ok && res.status !== 400) {
      if (![401, 403].includes(res.status)) {
        Sentry.captureMessage(
          `Request failed ${url} with status ${res.status}`,
          Sentry.Severity.Fatal,
        );
      } else {
        mutate('/restapi/v1/user/status');
      }
      return Promise.reject<RailsApiResponse<null>>({
        ...RailsApiResponseFallback,
        Code: res.status,
      });
    }
    return res.json();
  });
};

export const SwrFetcherConfig: ConfigInterface<
  any,
  any,
  (...args: any) => any
> = {
  fetcher: getApi,
  revalidateOnFocus: false,
  revalidateOnReconnect: false,
  errorRetryCount: 2,
  dedupingInterval: 30000,
  focusThrottleInterval: 10000,
  errorRetryInterval: 10000,
  onError: (err, key) => {
    if (![401, 403].includes(err.status)) {
      Sentry.captureMessage(
        `Request failed ${key} with status ${err.status}`,
        Sentry.Severity.Fatal,
      );
    }
  },
};

export const formatSuccesfullRailsApiResponse = <T>(
  data: T,
): RailsApiResponse<T> => ({
  Code: 0,
  Data: data,
  Message: '',
  Success: true,
});

export const API_VALIDATIONS = {
  email: async (email: string): Promise<RailsApiResponse<{}>> => {
    const res = await postApi<RailsApiResponse<{}>>(
      '/restapi/v1/registration/check/email',
      {
        email,
      },
    ).catch(err => err);
    return res;
  },
  personalCode: async (
    personal_code: string,
  ): Promise<RailsApiResponse<{}>> => {
    const res = await postApi<RailsApiResponse<{}>>(
      '/restapi/v1/registration/check/personal_code',
      {
        personal_code,
      },
    ).catch((err: RailsApiResponse<{}>) => err);
    return res;
  },
  postalCode: async (
    post_code: string,
  ): Promise<RailsApiResponse<RegistrationPostalCodeAutofill | null>> => {
    const res = await postApi<
      RailsApiResponse<RegistrationPostalCodeAutofill | null>
    >('/restapi/v1/registration/check/post_code', {
      post_code,
    }).catch((err: RailsApiResponse<null>) => err);
    return res;
  },
};
