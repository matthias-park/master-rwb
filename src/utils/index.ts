import { PagesName } from '../constants';
import { PageConfig } from '../types/api/PageConfig';
import Lockr from 'lockr';
import { PostItem } from '../types/api/Posts';
import dayjs from 'dayjs';

export const sortAscending = (a: number, b: number) => a - b;
export const sortDescending = (a: number, b: number) => b - a;

export function throttle(func: Function, limit: number): Function {
  let inThrottle: boolean;

  return function (this: any): any {
    const args = arguments;
    const context = this;
    if (!inThrottle) {
      inThrottle = true;
      func.apply(context, args);
      setTimeout(() => (inThrottle = false), limit);
    }
  };
}

export const stringToMiliseconds = (time: string): number | null => {
  const parsed = time.split(':');
  if (parsed.length === 3) {
    return miliseconds(Number(parsed[0]), Number(parsed[1]), Number(parsed[2]));
  }
  return null;
};
export const miliseconds = (
  hrs: number = 0,
  min: number = 0,
  sec: number = 0,
): number => {
  return (hrs * 60 * 60 + min * 60 + sec) * 1000;
};
export const removeFalsyFromObject = (obj: any) =>
  Object.keys(obj).forEach(
    k => !obj[k] && obj[k] !== undefined && delete obj[k],
  );

export const structuredBankCommunications = (barcode): string => {
  if (!barcode) return '';
  const rExp = /\d+/;
  const moduloOf97 = barcode.match(rExp)[0] % 97;
  return '00' + barcode + moduloOf97.toString().padStart(2, '0');
};

export const formatNavigationRoutes = (constants: PageConfig) => {
  const navigationRoutes = constants.navigation_routes
    ? [...constants.navigation_routes]
    : [];
  if (constants.content_pages) {
    for (const [key, value] of Object.entries(constants.content_pages)) {
      const name = Array.isArray(value) ? value[0] : value;
      const slug = Array.isArray(value) ? value[1] : key;
      const contentPagePath = `${slug.startsWith('/') ? '' : '/'}${slug}`;
      if (!navigationRoutes.some(route => route.path === contentPagePath)) {
        navigationRoutes.push({
          id:
            Array.isArray(value) && PagesName[key]
              ? Number(key)
              : PagesName.TemplatePage,
          path: contentPagePath,
          name,
        });
      }
    }
  }
  return navigationRoutes.sort((a, b) =>
    sortDescending(a.path.length, b.path.length),
  );
};

export const clearUserLocalStorage = () => {
  const keys = ['transactions-date-to', 'transactions-date-from'];
  for (const key of keys) {
    localStorage.removeItem(key);
  }
};

export const cleanPostBody = (obj: { [key: string]: unknown }) =>
  Object.entries(obj).reduce((obj, [key, value]) => {
    if (!key.includes('temp_field_') && value != null) {
      obj[key] = typeof value === 'string' ? value.trim() : value;
    }
    return obj;
  }, {});

export const errorHandler = (event: ErrorEvent) => {
  if (event.error?.stack?.includes('kambi-chunk')) {
    const kambiErrorId = 'kambi-error-reload';
    const kambiErrorRetryCount = Lockr.get(kambiErrorId, 0) + 1;
    if (kambiErrorRetryCount < 3) {
      Lockr.set(kambiErrorId, kambiErrorRetryCount);
      window.location.reload();
    }
  }
};

export const filterPromotionsList = (promotions: PostItem[]): PostItem[] =>
  promotions.filter(promotion => {
    const isNotExpired =
      !promotion.expiration_date ||
      dayjs(promotion.expiration_date).isAfter(dayjs());
    const isPublished =
      promotion.publish_date && dayjs(promotion.publish_date).isBefore(dayjs());
    const isVisible = promotion.visible;
    return isNotExpired && isPublished && isVisible;
  });
