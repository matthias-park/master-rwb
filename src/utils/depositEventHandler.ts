import { CustomWindowEvents, ComponentName } from '../constants';
import RequestReturn from '../types/api/deposits/RequestReturn';
import RailsApiResponse from '../types/api/RailsApiResponse';
import depositEventHandlerProps from '../types/DepositEventHandler';
import { postApi } from './apiUtils';
import * as Sentry from '@sentry/react';

const depositEventHandler = ({
  eventType,
  eventData,
  setCustomHtml,
  depositStatus,
  setApiError,
  setDepositLoading,
  t,
  depositRequestId,
  bankId,
  locale,
  setCurrentStep,
  disableModal,
  depositForm,
}: depositEventHandlerProps) => {
  switch (eventType) {
    case CustomWindowEvents.DepositPaymentSuccess: {
      setApiError(null);
      setCustomHtml(null);
      depositStatus.startCheckingStatus();
      break;
    }
    case CustomWindowEvents.DepositPaymentError: {
      if (typeof eventData === 'string') {
        Sentry.captureMessage(`deposit error: ${eventData}`);
        setApiError({ message: eventData, variant: 'danger' });
      } else {
        setApiError({ message: t('api_response_failed'), variant: 'danger' });
      }
      setCustomHtml(null);
      setDepositLoading(false);
      break;
    }
    case CustomWindowEvents.DepositPaymentWarning: {
      if (typeof eventData === 'string') {
        setApiError({ message: eventData, variant: 'warning' });
      }
      break;
    }
    case CustomWindowEvents.DepositRequestReturn: {
      if (eventData && typeof eventData === 'object') {
        const data: any = {
          ...eventData,
          depositRequestId,
        };
        if (bankId === 189) {
          data.browserInfo = {
            acceptHeader: '*',
            colorDepth: window.screen.colorDepth,
            javaEnabled: false,
            language: locale,
            screenHeight: window.innerHeight,
            screenWidth: window.innerWidth,
            timeZoneOffset: new Date().getTimezoneOffset(),
            userAgent: window.navigator.userAgent,
          };
        }
        postApi<RailsApiResponse<RequestReturn>>(
          '/restapi/v1/deposits/request_return',
          {
            bank_id: bankId.toString(),
            data: JSON.stringify(data),
          },
        )
          .then(res => {
            if (res.Data.OK) {
              if (res.Data.AdditionalData?.redirectUrl) {
                if (depositForm) {
                  window.open(res.Data.AdditionalData.redirectUrl, '_blank');
                  disableModal?.(ComponentName.QuickDepositModal);
                } else {
                  window.location.href = res.Data.AdditionalData.redirectUrl;
                }
                return;
              }
              if (res.Data.AdditionalData?.verifyPaymentData) {
                return window.postMessage(
                  JSON.stringify({
                    eventType: CustomWindowEvents.DepositVerifyPayment,
                    eventData: res.Data.AdditionalData.verifyPaymentData,
                  }),
                  '*',
                );
              }
              setCustomHtml(null);
              depositStatus.startCheckingStatus();
            } else if (res.Data.Error) {
              setCurrentStep?.(1);
              setApiError({
                message: t('api_response_failed'),
                variant: 'danger',
              });
              Sentry.captureMessage(
                `Got DepositRequestReturn error: ${res.Data.Error}`,
                Sentry.Severity.Info,
              );
            }
          })
          .catch(() => null);
      } else {
        Sentry.captureMessage(
          'Got DepositRequestReturn without data',
          Sentry.Severity.Fatal,
        );
      }
    }
  }
};
export default depositEventHandler;
