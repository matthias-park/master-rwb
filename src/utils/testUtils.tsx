import React from 'react';
import { render } from '@testing-library/react';
import { SWRConfig } from 'swr';
import { SwrFetcherConfig } from './apiUtils';
import { ToastProvider } from 'react-toast-notifications';
import { ConfigProvider } from '../hooks/useConfig';
import { I18nProvider } from '../hooks/useI18n';
import { HelmetProvider } from 'react-helmet-async';
import { UIConfigProvider } from '../hooks/useUIConfig';

const AllTheProviders = ({ children }) => {
  return (
    <SWRConfig value={SwrFetcherConfig}>
      <ToastProvider>
        <ConfigProvider>
          <I18nProvider>
            <HelmetProvider>
              <UIConfigProvider>{children}</UIConfigProvider>
            </HelmetProvider>
          </I18nProvider>
        </ConfigProvider>
      </ToastProvider>
    </SWRConfig>
  );
};

const customRender = (ui, options: any = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
