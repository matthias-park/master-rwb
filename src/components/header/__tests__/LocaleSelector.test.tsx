import React from 'react';
import { render, cleanup, fireEvent } from '../../../utils/testUtils';
import renderer from 'react-test-renderer';
import LocaleSelector from '../LocaleSelector';
import { act } from 'react-dom/test-utils';

beforeEach(cleanup);

const availableLocales = ['lt', 'en', 'de', 'fr'];
const current = 'en';

test('displays current Locale', async () => {
  const { getByTestId } = render(
    <LocaleSelector
      available={availableLocales}
      current={current}
      setLocale={() => {}}
    />,
  );
  expect(getByTestId('current')).toHaveTextContent(current);
});

test('displays mobile list', async () => {
  const { getByTestId } = render(
    <LocaleSelector
      available={availableLocales}
      current={current}
      setLocale={() => {}}
    />,
  );
  expect(getByTestId('available-mobile').children).toHaveLength(
    availableLocales.length,
  );
});
test('displays desktop list', async () => {
  const { getByTestId } = render(
    <LocaleSelector
      available={availableLocales}
      current={current}
      setLocale={() => {}}
    />,
  );
  await act(async () => {
    fireEvent.click(getByTestId('current'));
  });
  expect(getByTestId('available-desktop').children).toHaveLength(
    availableLocales.length,
  );
});

test('set new locale callback', async () => {
  let newLocale;
  const { getByTestId } = render(
    <LocaleSelector
      available={availableLocales}
      current={current}
      setLocale={locale => {
        newLocale = locale;
      }}
    />,
  );
  await act(async () => {
    fireEvent.click(getByTestId('current'));
  });

  await act(async () => {
    fireEvent.click(getByTestId('desktop-locale-fr'));
  });
  expect(newLocale).toBe('fr');
});

test('matches snapshot', () => {
  const dom = renderer.create(
    <LocaleSelector
      available={availableLocales}
      current={current}
      setLocale={() => {}}
    />,
  );
  expect(dom).toMatchSnapshot();
});
