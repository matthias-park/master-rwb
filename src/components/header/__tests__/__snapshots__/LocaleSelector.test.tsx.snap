// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="header__nav-item dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="header__nav-item-link d-none d-xl-block dropdown-toggle btn btn-link"
      data-testid="dropdown-trigger"
      disabled={false}
      id="navbarDropdownMenuLink"
      onClick={[Function]}
      type="button"
    >
      <strong
        className="text-uppercase"
        data-testid="current"
      >
        en
      </strong>
    </button>
  </div>,
  <li
    className="d-flex d-xl-none languages"
    data-testid="available-mobile"
  >
    <div
      className="lang-lt header__nav-item-link text-uppercase cursor-pointer"
      onClick={[Function]}
    >
      lt
    </div>
    <div
      className="lang-en header__nav-item-link text-uppercase cursor-pointer font-weight-bold"
      onClick={[Function]}
    >
      en
    </div>
    <div
      className="lang-de header__nav-item-link text-uppercase cursor-pointer"
      onClick={[Function]}
    >
      de
    </div>
    <div
      className="lang-fr header__nav-item-link text-uppercase cursor-pointer"
      onClick={[Function]}
    >
      fr
    </div>
  </li>,
]
`;
